controllers/create_addondeploymentconfig.go

```diff
@@ -34,10 +33,20 @@ func (r *SearchReconciler) createAddOnDeploymentConfig(ctx context.Context,
 		log.Error(err, "Error getting AddOnDeploymentConfig")
 		return &reconcile.Result{}, err
 	}
+	if !AddonDeploymentConfigEquals(found, adc) {
+		//get the current resource version to update
+		adc.ResourceVersion = found.ResourceVersion
```

Comment:
Copied over the resource version from existing AddOnDeploymentConfig as i was getting an error on update - Could not update AddOnDeploymentConfig - missing resourceversion. Please review. Thanks, @jlpadilla